File: backend/TradeX.Core/Models/BacktestKlineAnalysis.cs
using System.Text.Json.Serialization;

namespace TradeX.Core.Models;

public record BacktestTrade(
    int EntryIndex,
    int ExitIndex,
    DateTime EnteredAt,
    DateTime ExitedAt,
    decimal EntryPrice,
    decimal ExitPrice,
    decimal Quantity,
    decimal PnL,
    decimal PnLPercent);

public record BacktestKlineAnalysis(
    int Index,
    DateTime Timestamp,
    decimal Open,
    decimal High,
    decimal Low,
    decimal Close,
    decimal Volume,
    [property: JsonPropertyName("indicators")] Dictionary<string, decimal> IndicatorValues,
    [property: JsonPropertyName("entry")] bool? EntryConditionResult,
    [property: JsonPropertyName("exit")] bool? ExitConditionResult,
    bool InPosition,
    string Action,
    decimal? AvgEntryPrice = null,
    decimal? PositionQuantity = null,
    decimal? PositionCost = null,
    decimal? PositionValue = null,
    decimal? PositionPnl = null,
    decimal? PositionPnlPercent = null);

File: backend-go/internal/domain/backtest_kline_analysis.go
package domain

import (
	"time"

	"github.com/shopspring/decimal"
)

type BacktestKlineAnalysis struct {
	KlineIndex int
	Timestamp  time.Time

	Open   decimal.Decimal
	High   decimal.Decimal
	Low    decimal.Decimal
	Close  decimal.Decimal
	Volume decimal.Decimal

	IndicatorValues      map[string]float64
	EntryConditionResult *bool
	ExitConditionResult  *bool

	InPosition         bool
	Action             string // "enter" | "exit" | "hold"
	AvgEntryPrice      *decimal.Decimal
	PositionQuantity   *decimal.Decimal
	PositionCost       *decimal.Decimal
	PositionValue      *decimal.Decimal
	PositionPnl        *decimal.Decimal
	PositionPnlPercent *decimal.Decimal
}
